e2e/test/scenarios/models/reproductions.cy.spec.ts

```diff
@@ -1315,3 +1300,40 @@ describe("issue 46221", () => {
       .and("contain", modelDetails.name);
   });
 });
+
+describe("issue 20624", () => {
+  const questionDetails: StructuredQuestionDetails = {
+    name: "Question",
+    type: "question",
+    query: {
+      "source-table": PRODUCTS_ID,
+    },
+    visualization_settings: {
+      column_settings: {
+        '["name","VENDOR"]': { column_title: "Retailer" },
+      },
+    },
+  };
+
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+    cy.intercept("PUT", "/api/card/*").as("updateCard");
+  });
+
+  it("should reset the question's viz settings when converting to a model", () => {
+    createQuestion(questionDetails, { visitQuestion: true });
+    tableInteractive().within(() => {
+      cy.findByText("Retailer").should("be.visible");
+      cy.findByText("Vendor").should("not.exist");
+    });
+    openQuestionActions();
+    popover().findByText("Turn into a model").click();
+    modal().findByText("Turn this into a model").click();
+    cy.wait("@updateCard");
+    tableInteractive().within(() => {
+      cy.findByText("Vendor").should("be.visible");
+      cy.findByText("Retailer").should("not.exist");
+    });
```

Comment:
We should also assert that it's possible to edit column metadata (that's the symptom in original bug report).